File: backend/app/schemas/weather.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class WeatherCurrent(BaseModel):
    time: str
    temperature_2m: float | None
    relative_humidity_2m: float | None
    precipitation: float | None
    weather_code: int | None
    wind_speed_10m: float | None


class WeatherDay(BaseModel):
    date: str
    weather_code: int | None
    temp_min_c: float | None
    temp_max_c: float | None
    precipitation_sum_mm: float | None
    precipitation_probability_max: float | None
    et0_mm: float | None


class LocationWeatherForecast(BaseModel):
    location_id: UUID
    location_name: str
    latitude: float
    longitude: float
    timezone: str
    current: WeatherCurrent | None
    daily: list[WeatherDay]
    source: str
    fetched_at: datetime

File: frontend/src/types/weather.ts
export type WeatherCurrent = {
  time: string;
  temperature_2m: number | null;
  relative_humidity_2m: number | null;
  precipitation: number | null;
  weather_code: number | null;
  wind_speed_10m: number | null;
};

export type WeatherDay = {
  date: string;
  weather_code: number | null;
  temp_min_c: number | null;
  temp_max_c: number | null;
  precipitation_sum_mm: number | null;
  precipitation_probability_max: number | null;
  et0_mm: number | null;
};

export type LocationWeatherForecast = {
  location_id: string;
  location_name: string;
  latitude: number;
  longitude: number;
  timezone: string;
  current: WeatherCurrent | null;
  daily: WeatherDay[];
  source: string;
  fetched_at: string;
};
